backend/infrahub/graphql/analyzer.py

```diff
@@ -58,3 +323,201 @@ async def get_models_in_use(self, types: dict[str, Any]) -> set[str]:
+
+    def _get_operations(self) -> list[GraphQLQueryNode]:
+        operations: list[GraphQLQueryNode] = []
+        for operation_definition in self._operation_definitions:
+            selections = self._get_selections(selection_set=operation_definition.selection_set)
+
+            for field_node in selections.field_nodes:
+                schema_model: MainSchemaTypes
+                infrahub_node_models: list[NodeSchema] = []
+                model_name = self._get_model_name(node=field_node, operation_definition=operation_definition)
+
+                if model_name in self.schema_branch.node_names:
+                    schema_model = self.schema_branch.get_node(name=model_name, duplicate=False)
+                elif model_name in self.schema_branch.generic_names:
+                    schema_model = self.schema_branch.get_generic(name=model_name, duplicate=False)
+                    infrahub_node_models = [
+                        self.schema_branch.get_node(name=used_by, duplicate=False) for used_by in schema_model.used_by
+                    ]
+                elif model_name in self.schema_branch.profile_names:
+                    schema_model = self.schema_branch.get_profile(name=model_name, duplicate=False)
+                else:
+                    continue
+
+                operational_node = GraphQLQueryNode(
+                    operation=GraphQLOperation.from_operation(operation=operation_definition.operation),
+                    path=schema_model.kind,
+                    infrahub_model=schema_model,
+                    infrahub_node_models=infrahub_node_models,
+                    context_type=ContextType.from_operation(operation=operation_definition.operation),
+                    arguments=self._parse_arguments(field_node=field_node),
+                    variables=self._get_variables(operation=operation_definition),
+                )
+
+                if field_node.selection_set:
+                    selections = self._get_selections(selection_set=field_node.selection_set)
+                    for selection_field_node in selections.field_nodes:
+                        operational_node.children.append(
+                            self._populate_field_node(node=selection_field_node, query_node=operational_node)
+                        )
+                operations.append(operational_node)
+        return operations
+
+    @staticmethod
+    def _get_model_name(node: FieldNode, operation_definition: OperationDefinitionNode) -> str:
+        if operation_definition.operation == OperationType.MUTATION:
+            if node.name.value.endswith("Create"):
+                return node.name.value[:-6]
+            if node.name.value.endswith("Delete"):
+                return node.name.value[:-6]
+            if node.name.value.endswith("Update"):
+                return node.name.value[:-6]
+            if node.name.value.endswith("Upsert"):
+                return node.name.value[:-6]
```

Comment:
Maybe we can have something like:

```python
if node.name.value.endswith(("Create", "Delete", "Update", "Upsert")):
    return node.name.value[:-6]
```

But maybe it's also less explicit.

Reply:
Oh that part was more for the next step when I also want to add the requested action to specifically track if we need Create, Update or Delete permissions. So it will probably return more than a string at that point. Alternatively it might be a different function.